source/WanManager/wanmgr_policy_autowan_impl.c

```diff
@@ -2143,10 +2277,8 @@ static WcFmobPolicyState_t Transition_WanInterfaceUp(WanMgr_AutoWan_SMInfo_t *pS
     {
         if(pFixedInterface->VirtIfList->IP.ConnectivityCheckType == WAN_CONNECTIVITY_TYPE_TAD)
         {
-            if (WanMgr_GetWanServiceStatus() == 0)
-            {
-                wanmgr_setwanstart();
-            }
+            WanMgr_Sysevent_SetConnectivityCheckState(true, NET_CONNNECTIVITY_CHECK_STARTING);
+            wanmgr_setwanstart();
```

Comment:
since it's only for TAD, so there is very less possibility that any service wan running earlier, because all services trigger are only controlled by policy. other service wan triggered by any other component/places would be ignored by service wan.